lib/src/main/java/io/ably/lib/transport/ConnectionManager.java

```diff
@@ -835,10 +831,16 @@ public synchronized void connect() {
                 return;
             }
         }
+        if (closeRequested || currentState.terminal) {
+            // (RTN11d)
+            reinitializeChannelsAfterReconnect();
+            closeRequested = false;
+        }
         requestState(ConnectionState.connecting);
     }
 
     public void close() {
+        closeRequested = true;
```

Comment:
instead of `closeRequested`, can we use existing `cleaningUpAfterEnteringTerminalState = true`